### PR TITLE
go vet pkg

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

File: pkg/auth/server/login/implicit_test.go [hunk 1]
```diff
@@ -57,35 +57,35 @@ func TestImplicit(t *testing.T) {
 			},
 		},
 		"successful POST redirects": {
-			CSRF:       &csrf.FakeCSRF{"test", nil},
+			CSRF:       &csrf.FakeCSRF{Token: "test", Error: nil},
```
Comment (mfojtik) [inline pkg/auth/server/login/implicit_test.go]: is it more idiomatic to explicitly name keys?

**Comment by mfojtik** [comment]
awesome stuff @kargakis ! LGTM

File: pkg/auth/server/login/implicit_test.go (hunk 1 above)
Comment (kargakis) [inline pkg/auth/server/login/implicit_test.go]: go vet yells on unkeyed fields in struct literals so I guess it is.

There are still some more warnings though but I preffered not to touch them.
```
$ go vet pkg/
pkg/auth/oauth/handlers/authenticator_test.go:40: arg testCase.ExpectedAuthorized for printf verb %b of wrong type: bool
pkg/auth/oauth/registry/registry_test.go:286: github.com/GoogleCloudPlatform/kubernetes/pkg/util.Time composite literal uses unkeyed fields
pkg/auth/oauth/registry/registry_test.go:307: github.com/GoogleCloudPlatform/kubernetes/pkg/util.Time composite literal uses unkeyed fields
pkg/build/registry/etcd/etcd_test.go:21: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.EtcdHelper composite literal uses unkeyed fields
pkg/build/registry/etcd/etcd_test.go:21: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/build/webhook/github/github_test.go:239: arg proceed for printf verb %s of wrong type: bool
pkg/build/webhook/github/github_test.go:255: arg proceed for printf verb %s of wrong type: bool
pkg/build/webhook/github/github_test.go:279: arg proceed for printf verb %s of wrong type: bool
pkg/cmd/experimental/registry/registry.go:160: github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd.ClientConfigLoadingRules composite literal uses unkeyed fields
pkg/cmd/infra/router/router.go:94: unreachable code
pkg/cmd/server/admin/overwrite_bootstrappolicy.go:93: kubectl.ShortcutExpander composite literal uses unkeyed fields
pkg/cmd/server/etcd/etcd.go:80: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.EtcdHelper composite literal uses unkeyed fields
pkg/cmd/server/etcd/etcd.go:80: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/cmd/server/origin/master.go:543: unresolvable package for assetfs.AssetFS literal
pkg/cmd/server/origin/master.go:783: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.EtcdHelper composite literal uses unkeyed fields
pkg/cmd/server/origin/master.go:783: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/cmd/server/start/start_allinone.go:222: unreachable code
pkg/cmd/server/start/start_master.go:153: unreachable code
pkg/cmd/server/start/start_node.go:123: unreachable code
pkg/cmd/util/clientcmd/factory.go:66: github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl.ShortcutExpander composite literal uses unkeyed fields
pkg/config/cmd/cmd.go:35: github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.Mapper composite literal uses unkeyed fields
pkg/deploy/registry/etcd/etcd_test.go:56: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.EtcdHelper composite literal uses unkeyed fields
pkg/deploy/registry/etcd/etcd_test.go:56: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/image/registry/image/etcd/etcd_test.go:61: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/image/registry/imagerepository/etcd/etcd_test.go:27: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/image/registry/imagerepositorymapping/rest_test.go:28: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/image/registry/imagerepositorytag/rest_test.go:26: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/image/registry/imagestreamimage/rest_test.go:22: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/oauth/registry/etcd/etcd_test.go:20: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.EtcdHelper composite literal uses unkeyed fields
pkg/oauth/registry/etcd/etcd_test.go:20: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/oauth/server/osinserver/registrystorage/storage.go:159: github.com/GoogleCloudPlatform/kubernetes/pkg/util.Time composite literal uses unkeyed fields
pkg/oauth/server/osinserver/registrystorage/storage.go:199: github.com/GoogleCloudPlatform/kubernetes/pkg/util.Time composite literal uses unkeyed fields
pkg/route/registry/etcd/etcd_test.go:42: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.EtcdHelper composite literal uses unkeyed fields
pkg/route/registry/etcd/etcd_test.go:42: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/template/registry/etcd/etcd_test.go:18: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
pkg/user/registry/etcd/etcd_test.go:19: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.EtcdHelper composite literal uses unkeyed fields
pkg/user/registry/etcd/etcd_test.go:19: github.com/GoogleCloudPlatform/kubernetes/pkg/tools.RuntimeVersionAdapter composite literal uses unkeyed fields
```

**Comment by kargakis** [comment]
[test]

File: pkg/deploy/controller/configchange/controller.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func (c *DeploymentConfigChangeController) Handle(config *deployapi.DeploymentCo
 		_, _, err := c.generateDeployment(config)
 		if err != nil {
 			if kerrors.IsConflict(err) {
-				return fatalError(fmt.Sprintf("config %s updated since retrieval; aborting trigger", labelFor(config), err))
+				return fatalError(fmt.Sprintf("config %s updated since retrieval; aborting trigger", labelFor(config)))
```
Comment (smarterclayton) [inline pkg/deploy/controller/configchange/controller.go]: This should include the error message

File: pkg/cmd/util/variable/imagetemplate.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func NewDefaultImageTemplate() ImageTemplate {
 func (t *ImageTemplate) ExpandOrDie(component string) string {
 	value, err := t.Expand(component)
 	if err != nil {
-		glog.Fatalf("Unable to find an image for %q due to an error processing the format: %v", err)
+		glog.Fatalf("Unable to find an image for %s due to an error processing the format: %v", component, err)
```
Comment (smarterclayton) [inline pkg/cmd/util/variable/imagetemplate.go]: %q is intentional

File: pkg/build/webhook/generic/generic_test.go [hunk 4]
```diff
@@ -32,7 +32,7 @@ func TestVerifyRequestForMethod(t *testing.T) {
 	req := GivenRequest("GET")
 	err := verifyRequest(req)
 	if err == nil || !strings.Contains(err.Error(), "method") {
-		t.Errorf("Expected anything but POST to be an invalid method %s")
+		t.Errorf("Expected anything but POST to be an invalid method %s", err)
```
Comment (smarterclayton) [inline pkg/build/webhook/generic/generic_test.go]: should be %v

**Comment by kargakis** [comment]
@smarterclayton comments addressed

**Comment by kargakis** [comment]
re[test]

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**